operations/maintainer.go

```diff
@@ -195,6 +200,7 @@ func (om *Maintainer) CleanupFinishedCascadeOperations() {
 		query.ByField(query.InOperator, "state", string(types.SUCCEEDED), string(types.FAILED)),
 		// check if operation hasn't been updated for the operation's maximum allowed time to live in DB//
 		query.ByField(query.LessThanOperator, "updated_at", util.ToRFCNanoFormat(currentTime.Add(-om.settings.Lifespan))),
+		query.ByField(query.NotInOperator, "id", getObjectListIDs(lastOpsList)...),
```

Comment:
will this work on real systems with content? assuming all instances have last operations, this could be ~100k IDs, not sure postgres will like that..

Reply:
That's an excellent point! thanks Eyal!!!!
We did some research for a more efficient alternative and found that using NOT EXISTS should be far better performance wise.
see here:
https://explainextended.com/2009/09/16/not-in-vs-not-exists-vs-left-join-is-null-postgresql/
https://stackoverflow.com/questions/16996002/postgresql-query-is-slow-when-using-not-in
https://dba.stackexchange.com/questions/196020/postgresql-optimize-query-with-where-not-in-clause